shell/platform/embedder/tests/embedder_test_context_gl.cc

```diff
@@ -4,33 +4,93 @@
 
 #include "flutter/shell/platform/embedder/tests/embedder_test_context_gl.h"
 
+#include <memory>
 #include <utility>
 
 #include "flutter/fml/make_copyable.h"
 #include "flutter/fml/paths.h"
 #include "flutter/runtime/dart_vm.h"
 #include "flutter/shell/platform/embedder/tests/embedder_assertions.h"
 #include "flutter/shell/platform/embedder/tests/embedder_test_compositor_gl.h"
-#include "flutter/testing/test_gl_surface.h"
 #include "flutter/testing/testing.h"
 #include "tests/embedder_test.h"
 #include "third_party/dart/runtime/bin/elf_loader.h"
 #include "third_party/skia/include/core/SkSurface.h"
 
-namespace flutter {
-namespace testing {
+namespace flutter::testing {
 
 EmbedderTestContextGL::EmbedderTestContextGL(std::string assets_path)
     : EmbedderTestContext(std::move(assets_path)),
-      egl_context_(std::make_shared<TestEGLContext>()) {}
+      egl_context_(std::make_shared<TestEGLContext>()) {
+  renderer_config_.type = FlutterRendererType::kOpenGL;
+  renderer_config_.open_gl = {
+      .struct_size = sizeof(FlutterOpenGLRendererConfig),
+      .make_current = [](void* context) -> bool {
+        return reinterpret_cast<EmbedderTestContextGL*>(context)
+            ->GLMakeCurrent();
+      },
+      .clear_current = [](void* context) -> bool {
+        return reinterpret_cast<EmbedderTestContextGL*>(context)
+            ->GLClearCurrent();
+      },
+      .make_resource_current = [](void* context) -> bool {
+        return reinterpret_cast<EmbedderTestContextGL*>(context)
+            ->GLMakeResourceCurrent();
+      },
+      .fbo_reset_after_present = true,
+      .surface_transformation = [](void* context) -> FlutterTransformation {
+        return reinterpret_cast<EmbedderTestContextGL*>(context)
+            ->GetRootSurfaceTransformation();
+      },
+      .gl_proc_resolver = [](void* context, const char* name) -> void* {
+        return reinterpret_cast<EmbedderTestContextGL*>(context)
+            ->GLGetProcAddress(name);
+      },
+      .fbo_with_frame_info_callback =
+          [](void* context, const FlutterFrameInfo* frame_info) -> uint32_t {
+        return reinterpret_cast<EmbedderTestContextGL*>(context)
+            ->GLGetFramebuffer(*frame_info);
+      },
+      .present_with_info = [](void* context,
+                              const FlutterPresentInfo* present_info) -> bool {
+        return reinterpret_cast<EmbedderTestContextGL*>(context)->GLPresent(
+            *present_info);
+      },
+      .populate_existing_damage = nullptr,
+  };
+}
```

Comment:
Here and in the other `EmbedderTestContext` subclass ctors: this is just the code from the old `Initialize*FooRenderConfig` methods except using designated initialiser notation for readability, which also forces the members to be initialised in the order they're declared in embedder.h, hence the reordering.